### PR TITLE
less leak prone cleanup

### PR DESCRIPTION
**Pull request by dryajov**

### CODE REVIEW

File: libp2p/transports/tcptransport.nim [hunk 1]
```diff
@@ -77,12 +77,9 @@ proc connHandler*(t: TcpTransport,
       trace "Cleaning up client", addrs = $client.remoteAddress,
                                   conn
 
-      if not(isNil(conn) and conn.closed()):
-        await conn.close()
-
       t.clients[dir].keepItIf( it != client )
-      if not(isNil(client) and client.closed()):
-        await client.closeWait()
+      await allFuturesThrowing(
```
Comment (arnetheduck) [inline libp2p/transports/tcptransport.nim:81]: double all?

File: libp2p/protocols/secure/secure.nim [hunk 2]
```diff
@@ -77,8 +78,9 @@ proc handleConn*(s: Secure,
 
   proc cleanup() {.async.} =
     try:
-      await conn.join()
-      await sconn.close()
+      await conn.join() or sconn.join()
+      await allFuturesThrowing(
```
Comment (arnetheduck) [inline libp2p/protocols/secure/secure.nim:82]: double all?

File: libp2p/transports/tcptransport.nim (hunk 1 above)
Comment (dryajov) [reply libp2p/transports/tcptransport.nim:81]: heh, I keep tripping on this one not sure why.

Comment (arnetheduck) [reply libp2p/transports/tcptransport.nim:81]: too many magic helpers

Comment (dryajov) [reply libp2p/transports/tcptransport.nim:81]: true

File: libp2p/protocols/secure/secure.nim (hunk 2 above)
Comment (arnetheduck) [reply libp2p/protocols/secure/secure.nim]: I'm a bit worried about these "let's close as much as possible everywhere randomly" kind of approaches, they mask nondeterministic code flow paths that later make it harder to pinpoint races.. the interplay between the closeevent and who "owns" which connection becomes muddy

Comment (dryajov) [reply libp2p/protocols/secure/secure.nim]: Well, both tcptransport and secureconn have the particularity that they wrap some other resource, so when either the wrapped resource or the wrapper has been closed the other one needs to be closed as well. 

We can add additional checks to see which resource is already closed, to not call `close` unnecessarily,  but I've found that error prone and the checks are already done in the `close()` calls either way. I agree that it's not idea and we need more deterministic close flows, but I'm not sure how to make that happen with non deterministic object lifetimes and an async/event based flow.

**Pull request merged**